src/lib.rs

```diff
@@ -2103,6 +2102,68 @@ impl Url {
         Ok(())
     }
 
+    /// Return whether the URL is special.
+    ///
+    /// A URL is special if its scheme is one of the following:
+    ///
+    /// "http" | "https" | "ws" | "wss" | "ftp" | "gopher" | "file"
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use url::Url;
+    /// # use url::ParseError;
+    ///
+    /// # fn run() -> Result<(), ParseError> {
+    /// let url = Url::parse("ftp://rms@example.com")?;
+    /// assert!(url.is_special());
+    ///
+    /// let url = Url::parse("file://foo.bar")?;
+    /// assert!(url.is_special());
+    ///
+    /// let url = Url::parse("unix:/run/foo.socket")?;
+    /// assert!(!url.is_special());
+    ///
+    /// let url = Url::parse("data:text/plain,Stuff")?;
+    /// assert!(!url.is_special());
+    /// # Ok(())
+    /// # }
+    /// # run().unwrap();
+    /// ```
+    pub fn is_special(&self) -> bool {
```

Comment:
As I understand it, "special" is really about the scheme (which matches the docs), whereas a `Url::is_special()` method seems to be about the whole URL. Should this be renamed to `is_special_scheme()` or `is_scheme_special()` or `has_special_scheme()`?